File: QuickSort/Qsort.go
package main

import "fmt"

func Quicksort(arr []int) {
	if len(arr) <= 1 {
		return
	}

	pivot := arr[0] // Choose the first element as the pivot
	var left, right []int

	for _, num := range arr[1:] {
		if num <= pivot {
			left = append(left, num)
		} else {
			right = append(right, num)
		}
	}

	Quicksort(left)  // Recursively sort the left subarray
	Quicksort(right) // Recursively sort the right subarray

	copy(arr, append(append(left, pivot), right...)) // Combine the sorted subarrays and pivot
}

func main() {
	arr := []int{5, 2, 9, 1, 7, 6, 3}
	Quicksort(arr)
	fmt.Println(arr)
}

File: QuickSort/Qsort.py
def QuickSort(arr):
    if len(arr) <= 1:
        return arr
    pivot =arr[0]
    less = [i for i in arr[1:] if i <= pivot]
    greater =[i for i in arr[1:] if i > pivot]
    return QuickSort(less) + [pivot] + QuickSort(greater)
def main():
        arr = [64, 34, 25, 12, 22, 11, 90]
        qs = QuickSort(arr)
        print(qs)
if __name__ == "__main__":
    main()                    